csrc/scheduler/tools/domain_map.cpp

```diff
@@ -248,21 +247,30 @@ bool DomainMap::areAllTargetIdsCoveredBy(
   // https://github.com/NVIDIA/Fuser/issues/3542
 
   // Check all source iter domain involved in producing target_tv
-  for (IterDomain* source_id_out : get_source_iter_domains(target_tv)) {
+  for (IterDomain* source_id_out :
+       get_source_iter_domains(target_tv->getLogicalDomain())) {
     // NOTE: we use concrete id instead. This allows us to link indirect
     // broadcast. So in the example below: T2[i0, i1] = T0[i0, b0] + T1[i0, i1]
     // T3[i0, i9] = pad(T0[i0, b0])
     // We have i9 in T3
     //     -> source ID b0
     //     -> concrete map to i1
     // So T3 is contained by T2. See test `PointwiseTest.DomainMapPad1`
-    auto concrete_source_id_out =
+    auto concrete_id_out =
         ca_map_.getConcreteMappedID(source_id_out, IdMappingMode::PERMISSIVE);
-    // if we find any source_id_out that's not contained, it's possible our
-    // propagation would fail since transformation involving this iter domain
-    // can't be resolved.
-    if (!getMappedInputConcreteID(covered_source_ids, concrete_source_id_out)) {
-      return false;
+
+    // After mapping with PERMISSIVE map, `concrete_id_out` might no longer be a
```

Comment:
This is the patch. I added another call to `get_source_iter_domain({concrete_id_out})`

Reply:
So, for a given logical ID, we take its source IDs, then for each source ID, take the source IDs of its concrete ID. I understand it would work for the repro, but I wonder why it would need to take source IDs twice. For example, would it work if we took first the concrete ID of a logical ID and then its source IDs.

Reply:
I need to take the first `source ID` of the logical ID for the pad example to work, i.e. in the comment above from line 252 - 258.

```
T2[i0, i1] = T0[i0, b0] + T1[i0, i1]
T3[i0, i9] = pad(T0[i0, b0])
```

We want to have T2 as the reference TV and we want to be able to map T3 to T2. For i9 in T3. I wanted to first map to source ID, so it can be map to b0, which would concretize to i1.

So my naive justification of going through source IDs twice is that: the first source ID call allows us to trace back to broadcast IDs that went through a resize; while the second source ID calls allow us to resolve transformations on the concrete IDs.

Reply:
I'm fine with this PR as long as #3653 is fixed, but I'm feeling there's something not quite right here, probably because `T2` and `T3` shouldn't be actually mapped. They could have completely different sizes, and neither `T2` nor `T3` should not be considered representative.

Reply:
https://github.com/NVIDIA/Fuser/issues/3576#issuecomment-2540643470

Reply:
I agree with you.

We have some existing tests that asserts the behavior, which might not be what we would want to support moving forward.

I'll update the comment here per suggestion and merge this PR as-is just so we can patch the issue.

Meanwhile, I think the case where you are making is that, we shouldn't map through PERMISSIVE mapping in the first place here. I'll start a draft PR with that to see how many tests are affected. We can discuss what's the next step then.

Reply:
I'm not sure if it's something we should worry about at this moment. No test should fail functionally since it's a segmentation issue. It would be just a performance difference, but unless we have a benchmark that has this particular pattern, it doesn't make much sense to discuss what we should do.